File: wasm/sample/src/xplpc/message/request.ts
import { XPLPC } from "../core/xplpc";
import { Param } from "./param";

export class Request {
    functionName: string;
    params: Param[];

    constructor(functionName: string, ...params: Param[]) {
        this.functionName = functionName;
        this.params = params;
    }

    data(): string {
        return XPLPC.shared().config.serializer.encodeRequest(
            this.functionName,
            ...this.params,
        );
    }
}

File: flutter/plugin/lib/message/request.dart
import 'package:xplpc/core/xplpc.dart';
import 'package:xplpc/message/param.dart';

class Request {
  final String functionName;
  final List<Param>? params;

  Request(this.functionName, [this.params]);

  String data() {
    return XPLPC.instance.config.serializer.encodeRequest(functionName, params);
  }
}
